pkgs/development/compilers/vyper/default.nix

```diff
@@ -0,0 +1,69 @@
+{ stdenv, lib, python3, fetchPypi, git, python3Packages }:
+
+python3.pkgs.buildPythonPackage rec {
+  pname = "vyper";
+  version = "0.2.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "06qi1rmilrajf90lj1cn811j1dx4a6pdbri4m16cz2874r1wy9fq";
+  };
+
+
+  nativeBuildInputs = [ git ];
```

Comment:
Why is git needed? If we need to get a hash or something it is better to patch that.

Reply:
Here's what happens when you don't have git

```
$ nix-build -A vyper
these derivations will be built:
  /nix/store/p96ndzchmgx0f34pkw9s27wdb511gmfx-python3.8-vyper-0.2.7.drv
building '/nix/store/p96ndzchmgx0f34pkw9s27wdb511gmfx-python3.8-vyper-0.2.7.drv'...
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Using setuptoolsShellHook
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing setuptools-check-hook
Using setuptoolsCheckPhase
Sourcing pytest-check-hook
Using pytestCheckPhase
Removing setuptoolsCheckPhase
unpacking sources
unpacking source archive /nix/store/k6z1fr6qfp1g9nrzh30gfxnsp9cha8v4-vyper-0.2.7.tar.gz
source root is vyper-0.2.7
setting SOURCE_DATE_EPOCH to timestamp 1602731258 of file vyper-0.2.7/setup.cfg
patching sources
configuring
no configure script, doing nothing
building
Executing setuptoolsBuildPhase
Traceback (most recent call last):
  File "nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 41, in <module>
    commithash = subprocess.check_output("git rev-parse HEAD".split())
  File "/nix/store/0mabx9absir73mmijrsv0dsyvyya463r-python3-3.8.6/lib/python3.8/subprocess.py", line 411, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/nix/store/0mabx9absir73mmijrsv0dsyvyya463r-python3-3.8.6/lib/python3.8/subprocess.py", line 489, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/nix/store/0mabx9absir73mmijrsv0dsyvyya463r-python3-3.8.6/lib/python3.8/subprocess.py", line 854, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/nix/store/0mabx9absir73mmijrsv0dsyvyya463r-python3-3.8.6/lib/python3.8/subprocess.py", line 1702, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'git'
builder for '/nix/store/p96ndzchmgx0f34pkw9s27wdb511gmfx-python3.8-vyper-0.2.7.drv' failed with exit code 1
error: build of '/nix/store/p96ndzchmgx0f34pkw9s27wdb511gmfx-python3.8-vyper-0.2.7.drv' failed
```